src/main/java/Duke.java

```diff
@@ -50,11 +95,45 @@ public static void main(String[] args) {
                 }
                 System.out.println("\t" + taskList.get(doneTaskIndex).toString());
                 break;
+            case "todo":
+                if (inputArguments.length == REQUIRED_NUMBER_OF_ARGUMENT_FOR_TODO) {
+                    if (!description.isBlank()) {
+                        Todo newTodo = new Todo(description);
+                        addTask(taskList, newTodo);
+                        break;
+                    }
+                }
+
+                System.out.println(" Failed to add new todo: check syntax and no blanks!");
+                break;
+            case "deadline":
+                if (inputArguments.length == REQUIRED_NUMBER_OF_ARGUMENT_FOR_DEADLINE) {
+                    String byOption = inputArguments[2];
+                    String byInformation = inputArguments[3];
+                    if (!description.isBlank() && byOption.equals("/by") && !byInformation.isBlank()) {
+                        Deadline newDeadline = new Deadline(description, byInformation);
+                        addTask(taskList, newDeadline);
+                        break;
+                    }
+                }
+
+                System.out.println(" Failed to add new deadline: check syntax and no blanks!");
+                break;
+            case "event":
+                if (inputArguments.length == REQUIRED_NUMBER_OF_ARGUMENT_FOR_EVENT) {
+                    String atOption = inputArguments[2];
+                    String atInformation = inputArguments[3];
+                    if (!description.isBlank() && atOption.equals("/at") && !atInformation.isBlank()) {
+                        Event newEvent = new Event(description, atInformation);
+                        addTask(taskList, newEvent);
+                        break;
```

Comment:
Well done! You could refactor to check and break out of errors, and leave the main logic in the main indentation.

Reply:
Sure, I will see what I can do. Thank you.